### ISSUE TITLE
A Rust implementation?

### ISSUE THREAD

**Issue by teymour-aldridge** [supabase/pg-api#10]
I noticed that there is a Rust implementation of the Postgres client, but not for the server. I think that the server would benefit from the speed a Rust implementation (possibly more than the client) could bring.

**Comment by kiwicopple** [comment]
You're right about the client rust libraries. Bobbie has already started some of this work here: https://github.com/supabase/postgrest-rs
This library will get wrapped into a `supabase/supabase-rs` library which provides the broader functionality (sockets, auth etc).

As for re-writing the servers, Rust is very interesting for us and maybe one day there will be a re-write! Right now we want to build fast using languages which have massive community support. Node/Js has that feature (for better or worse), and we're a small team so we need a language which anyone in the team can jump into the code at any time. The realtime server is written in elixir for its concurrency and fault tolerance, so it's a unique case in our repos.

Also check out PostgREST: https://github.com/PostgREST/postgrest/. We use this heavily in Supabase but it's not one of our own repos, just a tool which we know and love. It's written in Haskell and Supabase wouldn't exist without it
